src/ExecutorContainer.ts

```diff
@@ -68,27 +79,13 @@ export default class ExecutorContainer extends EventEmitter implements Iterable<
 	 * @returns a new executor associated with the given language and file
 	 */
 	private createExecutorFor(file: string, language: LanguageId, needsShell: boolean) {
-		if (this.plugin.settings[`${language}Interactive`] || this.isAlwaysInteractive(language)) {
-			switch (language) {
-				case "js":
-					return new NodeJSExecutor(this.plugin.settings, file);
-				case "python":
-					return new PythonExecutor(this.plugin.settings, file);
-				case "prolog":
-					return new PrologExecutor(this.plugin.settings, file);
-			}
-		}
+		// Interactive language executor
+		if (this.plugin.settings[`${language}Interactive`])

```

Comment:
I know that you said if a language may be interactive, it must have an interactive executor. Still, I think it might be smart to add an `&& language in interactiveExecutors` check. I'm very open to discussion on this though :)

Reply:
I've just made it throw an error in the case that language is not in `interactiveExecutors` so that it's easier to spot if someone forgets to add it.